packages/hdwallet-xdefi/src/xdefi.ts

```
@@ -220,6 +228,14 @@ export class XDEFIHDWallet implements core.HDWallet, core.ETHWallet {
     return chainId === 1;
   }
 
+  public async ethGetChainId(): Promise<number> {
+    // at this point, we know that we're in the context of a valid XDEFI provider
+    const provider: any = await detectEthereumProvider({ mustBeMetaMask: false, silent: false, timeout: 3000 });
+    // chainId as hex string
+    const chainId: string = await provider.request({ method: "eth_chainId" });
+    return parseInt(chainId, 16);
```

I'm not sure what possible failure conditions exist for the async function calls here, but maybe this should be wrapped in a `try/catch` for safety.

---

Good point, I'm not sure about this one.
Since we've narrowed the provider itself to be valid, I would think anything going wrong with calling the Ethereum JSON-RPC provider with such method is really an edge case, what do you think? How would we handle the failed call?

---

The particular way we are using hdwallet in web guarantees that we'll be in the context of a valid XDEFI provider here, but hdwallet is used in other non-shapeshift projects. I'd abandon any assumptions about use case here and account for the cases where either of the await-ed calls fail. In this case, I'd throw an error, log the error to console, and change the return type to `Promise<number | null>`.

---

Added a try/catch in 75cc8e245128fe9827f15f0f3c37a98a507893e4 (note: I didn't throw since throwing would effectively make us unable to return anything else than void)